GliaWidgetsTests/Sources/CallViewModelTests.swift

```diff
@@ -181,4 +181,57 @@ class CallViewModelTests: XCTestCase {
         XCTAssertTrue(isLocalAudioStreamMuted)
         XCTAssertTrue(isLocalVideoStreamPaused)
     }
+    
+    func test_engagementTransferredReleasesRemoteAndLocalVideoAndShowsConnectingState() throws {
+        enum Calls {
+            case setLocalVideoNil
+            case setRemoteVideoNil
+            case showConnecting
+        }
+        var calls: [Calls] = []
+
+        var interactorEnv: Interactor.Environment = .failing
+        interactorEnv.gcd.mainQueue.asyncIfNeeded = { $0() }
+        let interactor: Interactor = .mock(environment: interactorEnv)
+        let mockOperator: CoreSdkClient.Operator = .mock()
+
+        call = .init(
+            .video(direction: .twoWay),
+            environment: .mock
+        )
+
+        viewModel = .init(
+            interactor: interactor,
+            alertConfiguration: .mock(),
+            screenShareHandler: ScreenShareHandler(),
+            environment: .mock,
+            call: call,
+            unreadMessages: .init(with: 0),
+            startWith: .engagement(mediaType: .video)
+        )
+
+        viewModel.action = { action in
+            switch action {
+            case .setLocalVideo(let video):
+                if video == nil {
```

Comment:
Why we need to check for `nil`?

Reply:
since we expect new streams from core, this just makes sure old ones are released from the view

Reply:
Can we be sure it is the case by using `XCTAssertNill(video)` instead of `if video == nil`?

Reply:
Updated tests